libraries/chain/controller.cpp

```diff
@@ -2708,8 +2708,8 @@ std::optional<block_id_type> controller::pending_producer_block_id()const {
 }
 
 const deque<transaction_receipt>& controller::get_pending_trx_receipts()const {
-   EOS_ASSERT( my->pending, block_validate_exception, "no pending block" );
-   return my->pending->get_trx_receipts();
+   static deque<transaction_receipt> empty;
+   return my->pending ? my->pending->get_trx_receipts() : empty;
```

Comment:
I would prefer to keep the assert to conform to other uses of `pending`.
Instead of this the caller should check `is_building_block()`.